### DISCUSSION TITLE
NSPanel bootloop after flashing Tasmota

### DISCUSSION THREAD

**Discussion by jancoow** [arendst/Tasmota#18315] [Q&A]
Hi,

My NSPanel is bricked after trying to flash Tasmota. It keeps booting with:

```
rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
invalid header: 0x00000001
invalid header: 0x00000001
invalid header: 0x00000001
```

I used esptool.py to flash it:

```
[janco@workpc Downloads]$ esptool.py write_flash -fm dio 0x0 tasmota32-nspanel.bin 
esptool.py v4.2-dev
Found 1 serial ports
Serial port /dev/ttyUSB25
Connecting.....................................
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision 3)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 24:d7:eb:0e:4b:18
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x00179fff...
Compressed 1547648 bytes to 1079819...
Wrote 1547648 bytes (1079819 compressed) at 0x00000000 in 95.4 seconds (effective 129.8 kbit/s)...
Hash of data verified.
```

Have flashed it several times, same problem.

Tried using the webUI, which doesn't work. Tried using Tasmotizer, but that tool things its an esp8266 for some reason.

**Comment by Noschvie** [comment]
Use [ESP Flasher](https://github.com/Jason2866/ESP_Flasher/releases/download/v.1.4.1/ESP-Flasher.exe) or the [Tasmota Web Installer](https://tasmota.github.io/install/) - flash Tasmota using a Chrome based browser for ESP82XX and ESP32
Are you using a recommended Serial Programmer?

**Comment by jancoow** [comment]
It seems that switching to my laptop it immediately work with using the webui.. Weird issue? I always use this desktop to flash my other esp12/esp32 devices with no problem at all.

**Reply by sfromis** [reply]
Well, it would only work if you used the correct binary. For first-time wired Tasmota flashing of ESP32, always use a `.factory.bin` file, and do not specify flash mode. Still, using the Tasmota Web Installer is much recommended, as this tool does the right thing.
